### PR TITLE
Remove the dependency of fetch on filemanager_thread::UIProvider.

### PR DESCRIPTION
**Pull request by Ms2ger**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14275)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Fails unit tests, hold on.

**Comment by Ms2ger** [comment]
Fixed.

File: components/net/filemanager_thread.rs [hunk 1]
```diff
@@ -112,26 +112,15 @@ enum FileImpl {
     Sliced(Uuid, RelativePos),
 }
 
-pub struct FileManager<UI: 'static + UIProvider> {
+#[derive(Clone)]
```
Comment (jdm) [inline components/net/filemanager_thread.rs]: Is this necessary?

**Review by jdm — CHANGES_REQUESTED** [review]

**Review by jdm — APPROVED** [review]

**Pull request merged**